test/unit/apps/gro/controllers/confirm.js

```diff
@@ -2,68 +2,216 @@
-    const callback = sinon.stub();
+  describe('formatStepsForEmail', () => {
+    let data;
+    beforeEach(() => {
+      data = controller.formatStepsForEmail({
+        'field-1': 'value-1',
+        'field-3': 'value-3',
+        'field-4': 'value-4',
+        'field-5': 'value-5',
+      });
+    });
+
+    it('returns an array with 2 entries', () => {
+      data.should.be.an('array').and.have.property('length').and.be.equal(2);
+    });
+
+    it('groups by section', () => {
+      data[0].should.have.property('section').and.be.equal('section-1');
+      data[1].should.have.property('section').and.be.equal('section-2');
+    });
+
+    describe('first section fields', () => {
+      it('is an array with 3 entries', () => {
+        data[0].fields.should.be.an('array').and.have.property('length').and.be.equal(3);
+      });
+
+      it('contains fields from section-1', () => {
+        data[0].fields[0].field.should.be.equal('field-1');
+        data[0].fields[1].field.should.be.equal('field-2');
+        data[0].fields[2].field.should.be.equal('field-3');
+      });
+
+      it('contains values passed for the fields', () => {
+        data[0].fields[0].value.should.be.equal('value-1');
+        expect(data[0].fields[1].value).to.be.undefined;
+        data[0].fields[2].value.should.be.equal('value-3');
+      });
+    });
+
+    describe('second section fields', () => {
+      it('is an array with 1 entries', () => {
+        data[1].fields.should.be.an('array').and.have.property('length').and.be.equal(1);
+      });
+
+      it('only contains field-4', () => {
+        data[1].fields[0].field.should.be.equal('field-4');
+      });
+    });
+  });
+
+  describe('saveValues', () => {
+    let data;
 
     beforeEach(() => {
-      controller = new ConfirmController({template: 'index'});
-      BaseController.prototype.saveValues = sinon.stub();
+      data = [{
+        fields: []
+      }];
+      callback = sinon.stub();
+      sinon.stub(Controller.prototype, 'formatStepsForEmail').returns(data);
+      sinon.stub(Controller.prototype, 'sendEmail').returns(promiseStub());
+    });
+
+    afterEach(() => {
+      Controller.prototype.formatStepsForEmail.restore();
+      Controller.prototype.sendEmail.restore();
+    });
+
+    it('calls callback with an error if formatStepsForEmail return no data', () => {
+      Controller.prototype.formatStepsForEmail.returns([]);
       controller.saveValues(req, res, callback);
+      callback.should.have.been.calledOnce.and.not.calledWithExactly();
+    });
+
+    it('adds a submission-date field to the first section', () => {
+      controller.saveValues(req, res, callback);
+      data[0].fields[0].should.have.property('field').and.be.equal('submission-date');
+    });
+
+    it('calls sendMail twice and callback with no error', done => {
+      controller.saveValues(req, res, err => {
+        Controller.prototype.sendEmail.should.have.been.calledTwice;
+        expect(err).to.be.undefined;
```

Comment:
should.exist(something); // not undefed or null
should.not.exist(somethingElse); // undefined or null
should.equal(somethingElse, undefined);